.github/workflows/test-desktop-reusable.yml

```diff
@@ -0,0 +1,433 @@
+      - name: upload diffs to s3
+        if: ${{ !cancelled() }}
+        uses: LedgerHQ/ledger-live/tools/actions/upload-images@develop
+        id: s3
+        with:
+          path: apps/ledger-live-desktop/tests/artifacts/test-results
+          workspace: ${{ github.workspace }}
+          os: linux
+          group-name: ${{ github.ref_name }}-${{ github.run_id }}-${{ github.run_number }}
+      - name: upload ci suggested screenshots
+        if: ${{ !cancelled() }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: images
+          path: images-linux.json
+      - name: Upload playwright test results [On Failure]
+        uses: actions/upload-artifact@v4
+        if: failure() && !cancelled()
+        with:
+          name: playwright-results-linux
+          path: |
+            apps/ledger-live-desktop/tests/artifacts/test-results
+            apps/ledger-live-desktop/tests/artifacts/html-report
+            apps/ledger-live-desktop/tests/artifacts/coverage
+            apps/ledger-live-desktop/tests/artifacts/videos
+            apps/ledger-live-desktop/tests/artifacts/logs
+            apps/ledger-live-desktop/tests/artifacts/*.log
+      - name: Upload Allure Report
+        if: ${{ !cancelled() }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: allure-results-linux
+          path: apps/ledger-live-desktop/allure-results
+  report:
+    needs: [codechecks, unit-tests, e2e-tests-linux]
+    runs-on: ubuntu-latest
+    if: ${{ !cancelled() }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.ref || github.sha }}
+      - name: "download linter results"
+        uses: actions/download-artifact@v4
+        with:
+          name: lint-desktop
+      - name: download images artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: images
+      - name: parse images
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const fs = require("fs");
+            const files = ["images-linux"];
+            let result = {};
+            for (const file of files) {
+              try {
+                const raw = JSON.parse(fs.readFileSync("${{github.workspace}}/" + file + ".json"));
+                const key = file.replace("images-", "").replace("-latest", "").trim()
+                result[key] = raw;
+              } catch (err) {
+                console.log(err);
+              }
+            }
+            fs.writeFileSync("./images.json", JSON.stringify(result, null, 2));
+      - name: prepare comment with screenshots
+        id: comment
+        uses: LedgerHQ/ledger-live/tools/actions/prepare-comment-screenshots@develop
+        with:
+          images: images.json
+          no-actor: true
+      - uses: actions/github-script@v7
+        name: prepare status
+        id: status
+        with:
+          script: |
```

Comment:
for easy maintainability of this script content, would it be bette to externalise it in a scripts directory under "workflows/" ? and just run it with it's path here ?

Reply:
Yes for most reusable workflows I only copy pasted what we had before. I think that we could do it in a dedicated PR and this is definitely an improvement we would like to have in coming weeks/monts.